### DISCUSSION TITLE
After Login Using Keycloak to Minio , Unable to View the Buckets

### DISCUSSION THREAD

**Discussion by NisharunnisaShaik** [minio/minio#13065] [Q&A]
Hi Team,

We deployed minio on our EKS & GKE Clusters. We also configured **Login with OpenID** using Keycloak 
(refernce: https://github.com/minio/minio/blob/master/docs/sts/keycloak.md) ,But after Logging into minio via OpenID we are unable to view any buckets, When we login with general credential of minio we are able to view all the buckets ,Please find the attached Screenshots for reference . 
Along with this we sometimes also facing issue of asking Client ID after clicking on the login with OpenID option , I already posted the same in Slack channel but couldnt get any help from there https://minio.slack.com/archives/C3NDUB8UA/p1626935403200600

Kindly let us know what we are doing wrong!

Thanks & Regards
Nisharunnisa

<img width="953" alt="After Keyclok" src="https://user-images.githubusercontent.com/57985230/130763450-b11eb40b-dabc-42ec-a23a-55ea21ac481d.PNG">

<img width="960" alt="Openid" src="https://user-images.githubusercontent.com/57985230/130763481-1f186a60-eede-4c2b-b5ab-e8997e4725f2.PNG">

<img width="223" alt="With minio" src="https://user-images.githubusercontent.com/57985230/130762764-fa5e9ea4-2379-4368-be65-270a1d34bb2c.PNG">

**Comment by vadmeste** [comment]
@NisharunnisaShaik usually generated credentials don't have any permissions to do anything. You need to associate a policy to a user claim

https://github.com/minio/minio/blob/master/docs/sts/keycloak.md

**Reply by NisharunnisaShaik** [reply]
Hi @vadmeste  I have created user in Keycloak to access minio , but still not able to access minio  buckets

**Reply by ravindk89** [reply]
@NisharunnisaShaik you need to look at our [documentation](https://docs.min.io/minio/baremetal/security/openid-external-identity-management/external-authentication-with-openid-identity-provider.html#access-control-for-externally-managed-identities)

Simply configuring external authentication is insufficient. You have to ensure that:

- The OIDC user has a claim that specifies the policy MinIO should attach, and
- That policy exists on MinIO

The documentation gives you the guidance you need to move forward.
